### ISSUE TITLE
software.sava.core.programs.Discriminator#equals() throws ArrayIndexOutOfBoundsException

### ISSUE THREAD

**Issue by imaykay** [sava-software/sava#4]
In
```
default boolean equals(final byte[] data, final int offset) {
    final byte[] thisData = data();
    final int len = data.length - offset;
    return len >= thisData.length && Arrays.equals(
        thisData, 0, thisData.length,
        data, offset, offset + data.length
    );
}
```
an ArrayIndexOutOfBoundsException is thrown when `Arrays.equals()` gets called . I guess the last param in the last row should be `offset + thisData.length`.

**Comment by jpe42** [comment]
can you provide an example? probably an easy fix.

**Comment by jpe42** [comment]
Nice catch. Fixed in 1.9.5 with commit 381627e.  Thanks working through this stuff.  feel free to share any top down use cases, I might be able to get ahead of these issues for you.

**Comment by jpe7s** [comment]
Nice catch. Fixed in 1.9.5 with commit https://github.com/sava-software/sava/commit/381627ef9d3c5e27807542389702ab1c0a274d0c. Thanks working through this stuff. feel free to share any top down use cases, I might be able to get ahead of these issues for you.
